apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainActivity.kt

```diff
@@ -54,17 +54,18 @@ class MainActivity : ReactActivity() {
           ){})
   }
 
-    // to do - add lifecycle methods
-//     override fun onResume() {
-//        super.onResume()
-//        MobileCore.setApplication(application);
-//        MobileCore.lifecycleStart(null);
-//    }
+    // Called when the main activity start interacting
+    override fun onResume() {
+        super.onResume()
+        (application as MainApplication).onResume()
+    }
+
+    // Called when the activity pauses
+    override fun onPause() {
+        super.onPause()
+        (application as MainApplication).onPause()
```

Comment:
Dito

Reply:
@cacheung  can you please discuss with Yangyansong and see what we can do as i was just using your code .

Reply:
have updated the code accordingly @yangyansong-adbe